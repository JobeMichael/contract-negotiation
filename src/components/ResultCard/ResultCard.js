import React from "react";
import currencyFormat from "../../helper/currencyFormatter";
import isValidNegotiation from "../../helper/isValidNegotiation";
import ModalContainer from "../ModalContainer/ModalContainer";
import WeatherCard from "../WeatherCard/WeatherCard";
import { Wrapper } from "./ResultCard.styles";

const ResultCard = ({ closeCardHandler, employerInput, employeeInput }) => {
  return (
    <ModalContainer setOpen={closeCardHandler}>
      <>
        <WeatherCard />

        <Wrapper>
          {isValidNegotiation(employerInput, employeeInput) ? (
            <h2 className="valid">Success</h2>
          ) : (
            <h2 className="failure">Failure</h2>
          )}
        </Wrapper>
        <p>
          Maximum Offer was <b>{currencyFormat(employerInput)}</b>
        </p>
        <p>
          Minimum expected salary was <b>{currencyFormat(employeeInput)}</b>
        </p>
      </>
    </ModalContainer>
  );
};

export default ResultCard;
