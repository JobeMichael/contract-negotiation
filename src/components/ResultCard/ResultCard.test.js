import { render } from "@testing-library/react";
import React from "react";
import ReactDOM from "react-dom";
import ResultCard from "./ResultCard";

describe("<ResultCard/>", () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  });

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  });

  it("renders ResultCard correctly", () => {
    const { asFragment } = render(
      <ResultCard employeeInput="2" employerInput="5" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders ResultCard with failure message", () => {
    const { getByText } = render(
      <ResultCard employeeInput="5" employerInput="2" />
    );
    const msg = getByText("Failure");

    expect(msg).toBeInTheDocument();
  });
});
