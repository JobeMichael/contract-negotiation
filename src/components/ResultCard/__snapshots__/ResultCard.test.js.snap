// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultCard/> renders ResultCard correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxhCb fSeUtm"
  />
  <div
    class="sc-AxjAm fuyJBj"
  >
    <div
      class="sc-AxhUy jkkAyo"
    >
      <a>
        Close
      </a>
    </div>
    <div
      class="sc-AxirZ kZUdNH"
    >
      <div
        class="sc-AxgMl eZymPG"
      >
        Loading weather...
      </div>
      <div
        class="sc-AxheI enektj"
      >
        <h2
          class="valid"
        >
          Success
        </h2>
      </div>
      <p>
        Maximum Offer was 
        <b>
          $ 5
        </b>
      </p>
      <p>
        Minimum expected salary was 
        <b>
          $ 2
        </b>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
